docs/gke/iap.md

```diff
@@ -0,0 +1,285 @@
+  * This URL will always be accessible regardless of whether IAP is enabled or not
+  * If IAP isn't enabled the app will report that you are an anyomous user
+  * Once IAP takes effect you will have to authenticate using your Google account and the app will tell you 
+    what your email is.
+  
+We also have the app running at
+
+```
+https://${FQDN}/whoami
+```
+  * But this path will reject traffic that didn't go through IAP so it won't work unless IAP is enabled. In this case you should get
+  one of the following errors.
+
+    * `401 UNAUTHORIZED` this indicates IAP isn't enabled so the request is rejected because it wasn't authenticated
+    * `401 ISS_AUD_UNMATCH1` this is because Envoy doesn't support IAP JWT tokens yet [istio/proxy/issues/941](https://github.com/istio/proxy/issues/941)
+
+
+### Disable JWT verification 
+
+Until [istio/proxy/issues/941](https://github.com/istio/proxy/issues/941) is fixed you can disable JWT verification in Envoy. 
+
+**Warning** This is a serious security risk because it means if IAP is disabled all traffic will be allowed through. You should only do this
+if you understand the risks; for more info see [IAP's docs on Securing Your App](https://cloud.google.com/iap/docs/signed-headers-howto).
+
+```
+ks param set iap-envoy disableJwtChecking true
+ks apply ${ENVIRONMENT} -c iap-envoy
+# Delete the pods so they are recreated with the new config
+kubectl delete pods --selector=service=envoy
+```
+
+### Configure Jupyter to use your Google Identity
+
+We can configure Jupyter to use the identity provided by IAP. This way users won't have to login to JupyterHub.
+
+```
+ks param set ${CORE_NAME} jupyterHubAuthenticator iap
+ks apply ${ENV} -c ${CORE_NAME}
+# Restart JupyterHub so it picks up the updated config
+kubectl delete -n ${NAMESPACE} pods tf-hub-0
+```
+  
+You should now be able to access JupyterHub at
+
+```
+https://${FQDN}/hub
+```
+
+### Adding Users
+
+IAP will block users who haven't been granted access to your web app. You can grant access using the following command.
+
+```
+gcloud projects add-iam-policy-binding $PROJECT \
+  --role roles/iap.httpsResourceAccessor \
+  --member user:${USER_EMAIL}
+```
+
+### Self signed certificates and browser security warnings
+
+Since you are using a self signed certificate chrome and other browsers will give you a warning like
+
+```
+Attackers might be trying to steal your information from ${ENDPOINT}(for example, passwords, messages, or credit cards). Learn more
+NET::ERR_CERT_AUTHORITY_INVALID
+```
+  * You will need to ignore these warnings
+  * To avoid these warnings you will need to use a certificate signed by a signing authority
+  * [Lets Encrypt](https://letsencrypt.org/) and other sites provide free signed certificates.
+
+
+## Troubleshooting
+
+Here are some tips for troubleshooting IAP.
+
+ * Make sure you are using https
+
+  ### 502 Server Error 
+    * A 502 usually means traffic isn't even making it to the envoy reverse proxy
```

Comment:
format looks incorrect because of unnecessary indent from here to the end of file.

Reply:
Thanks. PTAL.